sdk/storage/azure-storage-files-shares/src/share_file_client.cpp

```diff
@@ -202,18 +202,18 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
       const DownloadFileOptions& options) const
   {
     auto protocolLayerOptions = Details::ShareRestClient::File::DownloadOptions();
-    if (options.Offset.HasValue())
```

Comment:
You copied these changes from blob service, right? I didn't review this file since the logic here is complicated